sdk/nestjs/src/DevCycleModule/DevCycleService.ts

```diff
@@ -25,8 +24,8 @@ export class DevCycleService {
     variableValue<T extends DVCVariableValue>(
         key: string,
         defaultValue: T,
-    ): DVCVariableValue {
-        return this.devcycleClient.variableValue(
+    ): ReturnType<DevCycleClient['variableValue']> {
```

Comment:
is this correct without the generic being specified?

Reply:
(the returntype)

Reply:
Probably not... this is surprisingly annoying to get the correct return type here

Reply:
personally I would just disable that linting rule and let it infer here, assuming it compiles alright. Its usually better to not explicitly define it anyway since you run the risk of changing the type in subtle ways compared to what it should be

Reply:
Using the ReturnType with the generic would be preferable, but I haven't found a way to do that